packages/core/src/common/resource.ts

```diff
@@ -389,8 +403,8 @@ export class UntitledResource implements Resource {
     }
 }
 
-export function createUntitledURI(extension?: string, parent?: URI): URI {
-    const name = `Untitled-${untitledResourceSequenceIndex++}${extension ?? ''}`;
+export function createUntitledURI(extension?: string, parent?: URI, sequenceIndex?: number): URI {
```

Comment:
Instead of expanding on this `createUntitledURI` function, we should move it to the `UntitledResourceResolver` class (deprecate the existing function). We could then move the `do ... while` loop in here to remove the need for it in the `createUntitledResource` method and the execute function for `CommonCommands.NEW_FILE`.

Reply:
Thank you for reviewing at my code and your professional advice. I have already made changes to simplify.